crates/gitbutler-branch/src/reference.rs

```diff
@@ -0,0 +1,20 @@
+use gitbutler_reference::ReferenceName;
+use serde::{Deserialize, Serialize};
+
+use crate::BranchId;
+
+/// GitButler reference associated with a change (commit) on a virtual branch.
+/// These are not the same as regular Git references, but rather app-managed refs.
+/// Represent a deployable / reviewable part of a virtual branch that can be pushed to a remote
+/// and have a "Pull Request" created for it.
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+pub struct ChangeReference {
+    /// Branch id of the virtual branch this reference belongs to
+    /// Multiple references may belong to the same virtual branch, representing separate deployable / reviewable parts of the vbranch.
+    pub branch_id: BranchId,
+    /// Fully qualified reference name.
+    /// The reference must be a remote reference.
+    pub name: ReferenceName,
```

Comment:
the one that we had was kinda broken in subtle ways, so avoiding it here. perhaps this is a place where we could use gix types

Reply:
`gix` 'only' knows `gix::refs::FullName` which is a validated full name of any reference. From there, one can get the short-name as well, but I can't tell how practical it would be here.
I wouldn't fix what ain't broken, but will keep in mind that `ReferenceName` might also not be the most desirable type here.